EPRN.Portal/Services/PRNService.cs

```diff
@@ -224,11 +225,25 @@ public async Task<ViewPRNViewModel> GetViewPrnViewModel(string reference)
 
         public async Task<ActionPrnViewModel> GetActionPrnViewModel(int id)
         {
-            //TODO: This will need to be populated by the journey so we can return to it.
-            return new ActionPrnViewModel
-            {
-                Id = id
-            };
+            var reference = await _httpPrnsService.GetPrnReference(id);
+            var dto = await _httpPrnsService.GetPrnDetails(reference);
```

Comment:
Why don't you just create your own end point that returns the data that you need instead of making 2 calls? This is very inefficient

Reply:
Done.